pkg/networkserver/mac_adr_param_setup.go

```diff
@@ -60,23 +64,37 @@ func enqueueADRParamSetupReq(ctx context.Context, dev *ttnpb.EndDevice, maxDownL
 	}
 
 	if currentLimit == desiredLimit && currentDelay == desiredDelay {
-		return maxDownLen, maxUpLen, true, nil
+		return macCommandEnqueueState{
+			MaxDownLen: maxDownLen,
+			MaxUpLen:   maxUpLen,
+			Ok:         true,
+		}, nil
 	}
 
-	var ok bool
-	dev.MACState.PendingRequests, maxDownLen, maxUpLen, ok = enqueueMACCommand(ttnpb.CID_ADR_PARAM_SETUP, maxDownLen, maxUpLen, func(nDown, nUp uint16) ([]*ttnpb.MACCommand, uint16, bool) {
+	var st macCommandEnqueueState
+	dev.MACState.PendingRequests, st = enqueueMACCommand(ttnpb.CID_ADR_PARAM_SETUP, maxDownLen, maxUpLen, func(nDown, nUp uint16) ([]*ttnpb.MACCommand, uint16, []events.DefinitionDataClosure, bool) {
 		if nDown < 1 || nUp < 1 {
-			return nil, 0, false
+			return nil, 0, nil, false
 		}
 
 		req := &ttnpb.MACCommand_ADRParamSetupReq{
 			ADRAckLimitExponent: desiredLimit,
 			ADRAckDelayExponent: desiredDelay,
 		}
-		events.Publish(evtEnqueueADRParamSetupRequest(ctx, dev.EndDeviceIdentifiers, req))
-		return []*ttnpb.MACCommand{req.MACCommand()}, 1, true
+		log.FromContext(ctx).WithFields(log.Fields(
+			"ack_limit_exponent", req.ADRAckLimitExponent,
+			"ack_delay_exponent", req.ADRAckDelayExponent,
+		)).Debug("Enqueued ADRParamSetupReq")
+		return []*ttnpb.MACCommand{
+				req.MACCommand(),
+			},
+			1,
+			[]events.DefinitionDataClosure{
+				evtEnqueueADRParamSetupRequest.BindData(req),
+			},
+			true
```

Comment:
This looks a bit weird on my screen

Reply:
Putting it all on 1 line is even worse IMO.
I tried different layouts, but in the end I couldn't find a way to make it look nice - so I went for clarity. Please open a technical debt issue if you think it's necessary to do something about this.